runtimes/node-18.0/server.js

```diff
@@ -1,85 +1,138 @@
-        return send(res, 500, {stderr: 'Unauthorized'});
+    if (req.headers[`x-open-runtimes-secret`] !== process.env['OPEN_RUNTIMES_SECRET']) {
+        return send(res, 500, 'Unauthorized');
     }
 
-    const request = {
-        variables: body.variables ?? {},
-        headers: body.headers ?? {},
-        payload: body.payload ?? '',
-    };
-
-    console.stdlog = console.log.bind(console);
-    console.stderror = console.error.bind(console);
-    console.stdinfo = console.info.bind(console);
-    console.stddebug = console.debug.bind(console);
-    console.stdwarn = console.warn.bind(console);
-
     const logs = [];
     const errors = [];
+    const response = {
+        headers: {},
+        body: '',
+        statusCode: 200,
+    };
 
-    console.log = console.info = console.debug = console.warn = function() {
-        const args = [];
-        Array.from(arguments).forEach(arg => {
-            if(arg instanceof Object || Array.isArray(arg)) {    
-                args.push(JSON.stringify(arg));
-            } else {
-                args.push(arg)
-            }
-        });
-        logs.push(args.join(" "));
-    }
+    const contentType = req.headers['content-type'] ?? 'text/plain';
+    const rawBody = await parseText(req);
+    let body = rawBody;
 
-    console.error = function() {
-        const args = [];
-        Array.from(arguments).forEach(arg => {
-            if(arg instanceof Object || Array.isArray(arg)) {    
-                args.push(JSON.stringify(arg));
-            } else {
-                args.push(arg)
-            }
-        });
-        errors.push(args.join(" "));
+    if (contentType.includes('application/json')) {
+        body = await parseJson(req);
     }
 
-    const response = {
-        send: (text, status = 200) => send(res, status, {response: text, stdout: logs.join('\n'), stderr: errors.join('\n')}),
-        json: (json, status = 200) => send(res, status, {response: json, stdout: logs.join('\n'), stderr: errors.join('\n')}),
+    const headers = {};
+    Object.keys(req.headers).filter((header) => !header.startsWith('x-open-runtimes-')).forEach((header) => {
+        headers[header] = req.headers[header];
+    });
+
+    const context = {
+        req: {
+            rawBody,
+            body,
+            headers,
+            method: req.method,
+            url: req.url
+        },
+        res: {
+            send: function (body, statusCode, headers) {
+                if (body !== undefined) { response.body = body; }
+                if (statusCode !== undefined) { response.statusCode = statusCode; }
+                if (headers !== undefined) { response.headers = headers; }
+
+                return {
+                    body: response.body,
+                    statusCode: response.statusCode,
+                    headers: response.headers
+                }
+            },
+            json: function (obj, statusCode, headers = {}) {
```

Comment:
We need to set a default status code for json to be 200

Reply:
Added proper default values to all methods under `res`